changelog/unreleased/pull-2840

```diff
@@ -0,0 +1,6 @@
+Enhancement: Parallel file deletion in forget, prune and rebuild-index
```

Comment:
What about "Speed-up file deletion" which is slightly more focused on the observed behavior change?

Reply:
changed.